samples/performance/hpp_pipeline_cache/hpp_resource_record.cpp

```diff
@@ -0,0 +1,264 @@
+	{
+		recorder.set_graphics_pipeline(index, graphics_pipeline);
+	}
+};
+}        // namespace
+}        // namespace common
+}        // namespace vkb
+
+namespace vkb
+{
+namespace
+{
+inline void write_subpass_info(std::ostringstream &os, const std::vector<core::HPPSubpassInfo> &value)
+{
+	write(os, value.size());
+	for (const core::HPPSubpassInfo &item : value)
+	{
+		write(os, item.input_attachments);
+		write(os, item.output_attachments);
+	}
+}
+
+inline void write_processes(std::ostringstream &os, const std::vector<std::string> &value)
+{
+	write(os, value.size());
+	for (const std::string &item : value)
+	{
+		write(os, item);
+	}
+}
+}        // namespace
+
+void HPPResourceRecord::set_data(const std::vector<uint8_t> &data)
+{
+	stream.str(std::string{data.begin(), data.end()});
+}
+
+std::vector<uint8_t> HPPResourceRecord::get_data() const
+{
+	std::string str = stream.str();
+
+	return std::vector<uint8_t>{str.begin(), str.end()};
+}
+
+const std::ostringstream &HPPResourceRecord::get_stream()
+{
+	return stream;
+}
+
+size_t HPPResourceRecord::register_shader_module(vk::ShaderStageFlagBits stage, const ShaderSource &glsl_source, const std::string &entry_point, const ShaderVariant &shader_variant)
+{
+	LOGI("HPPResourceRecord::register_shader_module");
+
+	shader_module_indices.push_back(shader_module_indices.size());
+
+	write(stream, HPPResourceType::ShaderModule, stage, glsl_source.get_source(), entry_point, shader_variant.get_preamble());
+
+	write_processes(stream, shader_variant.get_processes());
+
+	return shader_module_indices.back();
+}
+
+size_t HPPResourceRecord::register_pipeline_layout(const std::vector<core::HPPShaderModule *> &shader_modules)
+{
+	LOGI("HPPResourceRecord::register_pipeline_layout");
+
+	pipeline_layout_indices.push_back(pipeline_layout_indices.size());
+
+	std::vector<size_t> shader_indices(shader_modules.size());
+	std::transform(shader_modules.begin(), shader_modules.end(), shader_indices.begin(),
+	               [this](core::HPPShaderModule *shader_module) { return shader_module_to_index.at(shader_module); });
+
+	write(stream,
+	      HPPResourceType::PipelineLayout,
+	      shader_indices);
+
+	return pipeline_layout_indices.back();
+}
+
+size_t HPPResourceRecord::register_render_pass(const std::vector<rendering::HPPAttachment> &attachments, const std::vector<common::HPPLoadStoreInfo> &load_store_infos, const std::vector<core::HPPSubpassInfo> &subpasses)
+{
+	LOGI("HPPResourceRecord::register_render_pass");
+
+	render_pass_indices.push_back(render_pass_indices.size());
+
+	write(stream,
+	      HPPResourceType::RenderPass,
+	      attachments,
+	      load_store_infos);
+
+	write_subpass_info(stream, subpasses);
+
+	return render_pass_indices.back();
+}
+
+size_t HPPResourceRecord::register_graphics_pipeline(vk::PipelineCache /*pipeline_cache*/, rendering::HPPPipelineState &pipeline_state)
+{
+	LOGI("HPPResourceRecord::register_graphics_pipeline");
```

Comment:
Maybe you could remove all those logging calls to make the hpp-version as fast as the non-hpp-version.